Library/Homebrew/github_runner_matrix.rb

```diff
@@ -181,7 +199,13 @@ def generate_runners!
       )
       @runners << create_runner(:macos, :arm64, spec, macos_version)
 
-      next if !@all_supported && macos_version > NEWEST_HOMEBREW_CORE_INTEL_MACOS_RUNNER
+      # `#deploy_new_x86_64_runner?` is expensive, so:
+      #  - avoid calling it if we don't have to
+      #  - cache the result to a variable to avoid calling it multiple times
+      if macos_version > NEWEST_DEFAULT_HOMEBREW_CORE_INTEL_MACOS_RUNNER &&
```

Comment:
Was removing the `!@all_supported` intentional?

Reply:
It's up here: https://github.com/Homebrew/brew/blob/30e25c540372fd0394ee2c8e0dbf06651cc125f1/Library/Homebrew/github_runner_matrix.rb#L63

Reply:
Ah thanks, that wasn't obvious.